libs/template/helpers.go

```diff
@@ -27,4 +32,23 @@ var helperFuncs = template.FuncMap{
 	"regexp": func(expr string) (*regexp.Regexp, error) {
 		return regexp.Compile(expr)
 	},
+	// A key value pair. This is used with the map function to generate maps
+	// to use inside a template
+	"pair": func(k string, v any) pair {
+		return pair{k, v}
+	},
+	// map converts a list of pairs to a map object. This is useful to pass multiple
+	// objects to templates defined in the library directory. Go text template
+	// syntax for invoking a template only allows specifying a single argument,
+	// this function can be used to workaround that limitation.
+	//
+	// For example: {{template "my_template" (map (pair "foo" $arg1) (pair "bar" $arg2))}}
+	// $arg1 and $arg2 can be referred from inside "my_template" as ".foo" and ".bar"
+	"map": func(pairs ...pair) map[string]any {
```

Comment:
can we use `(map ("key" "value))` syntax instead? You can take a look at possible implementation here

https://github.com/databricks/databricks-sdk-go/blob/main/openapi/code/tmpl_util_funcs.go#L62

Reply:
This approach does not allow demarkation of key value pairs by using `()`, which is important here. This is because the otherwise in `("key" value)` `"key"` is treated as an function and template parsing errors.